### ISSUE TITLE
RAM Usage

### ISSUE THREAD

**Issue by marek9499** [niklasr22/BrightIntosh#156]
hey, today I bought the application and I am very happy with it, although I noticed that it consumes quite a lot of RAM - just like messenger. Taking into account the point of the application, I don't know what to call it - the issue of its simplicity, I think that the application should not take up so much RAM. In this application we have fabulously simple components such as a slider, window and buttons, and indeed mechanics - logic. For example: Messenger has real-time socket communication, audio/video connections, encryption and many others. I'm sending a screenshot from the app's process manager for comparison. I have an M3 Max with 36GB RAM and I have a large memory reserve, but there are people who have 16 and even 8GB RAM. I would send a pull request, but I develop in TypeScript, not Swift :D Therefore, dear author, could you pay attention to what is draining RAM in your app? And if it doesn't work, let me know and I'll try to check something myself.

<img width="920" alt="Zrzut ekranu 2024-06-28 o 13 19 11" src="https://github.com/niklasr22/BrightIntosh/assets/32765296/ded46047-8d82-4fc2-a0e5-854181664916">

**Comment by niklasr22** [comment]
Hi, thanks for reaching out.  For me the application only requires about 30-45MB. I agree that 350MB is way too much.
![Screenshot 2024-06-28 at 14 13 04](https://github.com/niklasr22/BrightIntosh/assets/75939868/02318c64-27dc-4cc9-ae7a-e4bfe68820fb)

Could you please check which version of BrightIntosh you are running and which settings you have enabled?

**Comment by marek9499** [comment]
I will say this - after turning on the application, the amount of RAM is actually about 30-50MB, although after turning on additional brightness it jumps to 330MB. Is it the same for you?

https://github.com/niklasr22/BrightIntosh/assets/32765296/731fa6dc-a626-47d2-abd9-7dbef8f31846

**Comment by niklasr22** [comment]
Thanks for the video. Unfortunately I can't reproduce the issue like this on my devices.

**Comment by niklasr22** [comment]
Maybe it has something to do with the generation of the MacBook. I have only M1 based devices, but I was said that on an M3 Pro BrightIntosh allocates 160MB of RAM...

**Comment by marek9499** [comment]
of course, thanks for the info - I'll check it next week. I'm currently on vacation and don't have access to a Mac 😁

**Comment by diego351** [comment]
Just checked on my M2 pro 14 inch MacBook Pro:

<img width="743" alt="image" src="https://github.com/niklasr22/BrightIntosh/assets/1809191/e9849116-e118-49e9-8c97-2ad5484f8305">

**Comment by marek9499** [comment]
I'm back from vacation and: when the app is turned on and the brightness is deactivated, the RAM usage is 30-50MB, which seems to be OK. When I click activate from the bar menu (next to the time), RAM Usage is 330-400MB.

**Comment by marek9499** [comment]
this seems a bit unbelievable - can you record your screen and show me that the RAM stays the same after turning on extra brightness?

**Comment by diego351** [comment]
I think you should provide video if any, your the one reporting the issue

**Comment by UnluckyY1** [comment]
Tested on M3 PRO device 
<img width="961" alt="image" src="https://github.com/user-attachments/assets/2fbbe7c7-81dc-4d15-8319-aa63f3639e91">

**Comment by marek9499** [comment]
What a surprise! so I'm not the only one who has this error where the application "eats" RAM. Can I make a memory dump of the application so you can figure out what is taking up memory?

**Comment by niklasr22** [comment]
This screenshot shows what I meant, the memory column shows a much larger value than the Real Memory column. Unfortunately it is very hard to find official documentation on what exactly those values describe. XCode provides different memory inspection tools: The integrated [memory report](https://developer.apple.com/documentation/xcode/gathering-information-about-memory-use) and Allocation information in the Instruments app (also described in the article). Those tools allow an export of memory graphs and traces. Unfortunately the values of both tools don't match each other and don't match the values shown in the Activity Monitor.
